compiler/lib/src/main/scala/analysis/CheckSemantics/CheckUses.scala

```diff
@@ -117,6 +117,33 @@ object CheckUses extends UseAnalyzer {
     yield a.copy(nestedScope = a.nestedScope.pop)
   }
 
+  override def defTopologyAnnotatedNode(a: Analysis, node: Ast.Annotated[AstNode[Ast.DefTopology]]) = {
+    a.dictionaryMode match {
+      case true => {
+        val impliedTypeUses = List(
+          "FwChanIdType", 
+          "FwEventIdType", 
+          "FwOpcodeType", 
+          "FwPacketDescriptorType", 
+          "FwTlmPacketizeIdType"
+        )
+        val (_, node1, _) = node
+        val mapping = a.nestedScope.get (NameGroup.Type) _
+        for {
+          a <- Result.foldLeft (impliedTypeUses) (a) ((a, t) => {
+            for {
+              symbol <- Result.annotateResult(
+                helpers.getSymbolForName(mapping)(node1.id, t), 
+                s"Dictionary requires type ${t} to be defined.")
```

Comment:
Let's add another check here to ensure that the type has an integer type as its underlying type. If not, we can return a TypeError saying that an integer type is required.

Reply:
Added this check to FinalizeTypeDefs.